### DISCUSSION TITLE
Once I've set up a socket.io connection between server.js and client.js, is it possible to also connect to an external websocket API, and broadcast it to the internal connection?

### DISCUSSION THREAD

**Discussion by respectabiggle** [socketio/socket.io#4490] [Q&A] [answered]
My app currently connects to the same external ws URL twice, once from server.js and once from client.js. This seems inefficient.

    example: wss://stream.example.com:5555/ws/GMT4@weatherdata

Is it possible to pass an externally-sourced data stream between server.js and client.js with socket.io?

This [StackOverflow thread](https://stackoverflow.com/questions/70439903/how-to-connect-to-a-external-websocket-using-node-js) implies that this is not possible with socket.io.

Am I misreading the thread, or is it mistaken?

Thank you.

**Comment by darrachequesne** [comment, marked as answer]
Hi! You should be able to pipe the events from the source to the Socket.IO connection:

```js
import { Server } from "socket.io";
import WebSocket from "ws";

const io = new Server(3000);

const client = new WebSocket("wss://stream.example.com:5555/ws/GMT4@weatherdata");

client.on("message", (data) => {
  io.emit("weatherdata", data);
});

// TODO handle reconnection to the source
```

Does it suit your use case?

**Reply by respectabiggle** [reply]
Thank you!
My weekend is over -- I will try this in a couple days and report back.
Thanks again for your reply!!

*** 

Later:

I can't express enough gratitude.

Your example works.
But what's more: it helped me understand better and follow along with the socket.io docs (I am new), to keep developing my feature.

I am proud of myself, but also very thankful to you for being so friendly!

Best wishes and thanks again.
